application/common/application_file_util.cc

```diff
@@ -170,7 +168,8 @@ inline bool IsElementSupportSpanAndDir(xmlNode* root) {
 }
 
 bool IsSingletonElement(const std::string& name) {
-  for (int i = 0; i < arraysize(kSingletonElements); ++i)
```

Comment:
Huh? It looks like you didn't understand either what the `arraysize()` macro does or what is wrong with the original code. Just change `i`'s type to `size_t`.